mods/ra/maps/production-disruption/production-disruption.lua

```diff
@@ -0,0 +1,270 @@
+		Actor.Create("4tnk", true, { Owner = ussr, Facing = 127, Location = Mammoth.Location })
+		Actor.Create("4tnk", true, { Owner = ussr, Facing = 127, Location = Mammoth.Location + CVec.New(1,0) })
+		Actor.Create("v2rl", true, { Owner = ussr, Facing = 127, Location = V2.Location })
+	end
+
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Chalk2.TargetParatroopers(LZ.CenterPosition, Angle.New(780))
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		UnitsArrived = true
+		TeslaCamera = Actor.Create("camera", true, { Owner = greece, Location = TeslaCam.Location })
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(10), function()
+		LZCamera.Destroy()
+		Utils.Do(RifleSquad1, function(actor)
+			if not actor.IsDead then
+				actor.AttackMove(LZ.Location)
+				actor.Hunt()
+			end
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(15), function()
+		TeslaCamera.Destroy()
+		Utils.Do(RifleSquad2, function(actor)
+			if not actor.IsDead then
+				actor.AttackMove(LZ.Location)
+				actor.Hunt()
+			end
+		end)
+	end)
+end
+
+SetupTriggers = function()
+	Trigger.OnDamaged(SubPen, function()
+		Utils.Do(Heavys, function(actor)
+			if not actor.IsDead then
+				actor.Hunt()
+			end
+		end)
+	end)
+
+	Trigger.OnKilled(Church, function()
+		Actor.Create("healcrate", true, { Owner = greece, Location = ChurchCrate.Location })
+	end)
+
+	Trigger.OnKilled(ObjectiveDome, function()
+		if not DomeCaptured == true then
+			greece.MarkFailedObjective(CaptureDome)
+		end
+	end)
+
+	Trigger.OnKilledOrCaptured(SubPen, function()
+		greece.MarkCompletedObjective(StopProduction)
+	end)
+end
+
+PowerDown = false
+PowerDownTeslas = function()
+	if not PowerDown == true then
+		CaptureDome = greece.AddSecondaryObjective("Capture the enemy radar dome.")
+		greece.MarkCompletedObjective(PowerDownTeslaCoils)
+		Media.PlaySpeechNotification(greece, "ReinforcementsArrived")
+		PowerDown = true
+		local demoEngis = Reinforcements.ReinforceWithTransport(greece, LSTType, DemoEngiTeam, DemoEngiPath, { DemoEngiPath[1] })[2][1]
+		Trigger.OnAddedToWorld(demoEngis, function()
+			CaptureRadarDome()
+		end)
+
+		local bridge = Utils.Where(Map.ActorsInWorld, function(actor) return actor.Type == "bridge1" end)[1]
+		if not bridge.IsDead then
+			bridge.Kill()
+		end
+
+		FlameTowersCam = Actor.Create("camera", true, { Owner = greece, Location = FlameCam.Location })
+		Trigger.AfterDelay(DateTime.Seconds(10), function()
+			FlameTowersCam.Destroy()
+		end)
+	end
+end
+
+DomeCaptured = false
+CaptureRadarDome = function()
+	Trigger.OnCapture(ObjectiveDome, function()
+		DomeCaptured = true
+		greece.MarkCompletedObjective(CaptureDome)
+		SendChronos()
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Engineers = Utils.Where(greece.GetActors(), function(actor)
+			return
+				actor.Type == "e6"
+		end)
+
+		Trigger.OnAllRemovedFromWorld(Engineers, function()
+			if not DomeCaptured == true then
+				greece.MarkFailedObjective(CaptureDome)
```

Comment:
We need to fail the mission here in case the player hasn't repaired the bridge, as there is no way to progress.